npm/private/npm_link_package_store.bzl

```diff
@@ -126,6 +126,8 @@ def _impl(ctx):
             runfiles = ctx.runfiles(transitive_files = transitive_files_depset),
         ),
         js_info(
+            # assume a directory contains declarations since we can't know that it doesn't
+            declarations = files_depset,
```

Comment:
Should this be `virtual_store_directory` instead of `files_depset`?

The `files_depset` is a symlink to the `virtual_store_directory` but possibly also includes both the file and the symlink (based on `experimental_allow_unresolved_symlinks`?).

Reply:
`files_depset` is right. You want both the node_modules/foo symlink and the underlying virtual store foo directory

Reply:
depending on `experimental_allow_unresolved_symlinks` `ctx.actions.symlink` is either called with `target_path` or `target_file`. If `target_file` is used then the virtual store is included transitively in the symlink. When using `target_path` then underlying virtual store directory needs to be included explicitly along with the symlink File

Reply:
technically you should also set `transitive_declarations` to `transitive_files_depset`

Reply:
> technically you should also set transitive_declarations to transitive_files_depset

I guess just like the npm package itself we don't know what's in the transitive files so we should just include it 100% of the time? Otherwise it won't be available to the ts compiler and might fail?

Reply:
Yup. I don't know if any rules even use the `transitive_declarations` but they are there incase they are ever needed